packages/date-picker/src/shared/components/DateInput/DateInputSegment/DateInputSegment.tsx

```diff
@@ -95,10 +95,17 @@ export const DateInputSegment = React.forwardRef<
 
     /** Handle keydown presses that don't natively fire a change event */
     const handleKeyDown: KeyboardEventHandler<HTMLInputElement> = e => {
-      const { key } = e as React.KeyboardEvent<HTMLInputElement> & {
+      const { key, target } = e as React.KeyboardEvent<HTMLInputElement> & {
         target: HTMLInputElement;
       };
 
+      // if a number is pressed and the value length is equal to the charsPerSegment, reset the input
+      if (isFinite(Number(key))) {
```

Comment:
Huh, I've never seen this check. Why do we need to check if the key is finite?

Reply:
I'll change this. This should be a check for only numbers.

Reply:
is there a case where `key` is not a number? (I guess maybe it could be undefined?)

Reply:
Yeah, it could be one of the arrow keys, enter, space, etc...